### DISCUSSION TITLE
Use of yarpmanager within assignment_make-it-roll

### DISCUSSION THREAD

**Discussion by accateo** [robotology/community#216] [Q&A] [answered]
Hi,

I've come up with my solution to the [`make-it-roll` assignment][1] and I want to test it using `yarpmanager`. Thus, I have launched `yarpmanager` and imported the two applications available in the app folder.

The `assignment_make-it-roll-app` app works just fine, whereas the `assignment_make-it-roll-system` app does not. In fact, only the `yarplogger` gets started correctly, while the other modules go in timeout waiting for resources.

What resources do I need?

Thanks

[1]: https://github.com/vvv-school/assignment_make-it-roll

**Comment by pattacini** [comment]
@Arya07 @Iaxama @gmmarconi @massimoregoli @towardthesea @giuliavezzani 
anyone willing to help?

**Comment by Iaxama** [comment, marked as answer]
While the yarplogger does not have any dependency, all of the other modules depend on the iCubSim to be up and running to start. Try to run iCubSim from a terminal and check that you can run it properly, then try again.

**Comment by accateo** [comment]
Ok, now it works. I have also sent rpc command to the robot, in this way the robot look down.
Now the robot should search the blue ball or I have to move the head to the right?

**Comment by Iaxama** [comment]
You can use the getCog method provided in the main.cpp to get the center of the ball in the two images. Then you have to write the code to fixate at that point (Hint: use the IGazeControl to do that)

**Comment by accateo** [comment]
No ok, I already use getCog and IGazeControl in the code, but when I run it the robot looks forward. 
I run yarpmanager with the two modules, yarpserver and the compiled main.cpp code

**Comment by Iaxama** [comment]
For the getCog method to work the ball should be in the visual range of the cameras. The robot has to look down towards the table first, then get the ball position in the two images with the getCog method, then you need to retrieve the ball position using the iGazeControl.triangulate3dPoint method, then ask the robot to approach the ball position with the hand and push it.

**Comment by accateo** [comment]
Ok, thanks!

**Comment by pattacini** [comment]
Closing for now; feel free to open it again if needed.
